File: testprojects/streaming/python/pipelines/JSONType/code/jsontype/graph/WindowFunction_1.py
from pyspark.sql import *
from pyspark.sql.functions import *
from pyspark.sql.types import *
from jsontype.config.ConfigStore import *
from jsontype.udfs.UDFs import *

def WindowFunction_1(spark: SparkSession, in0: DataFrame) -> DataFrame:
    return in0\
        .withColumn("p_string", row_number().over(Window.partitionBy(col("p_int")).orderBy(col("p_float").asc())))\
        .withColumn("c_tinyint", row_number().over(Window.partitionBy(col("p_int")).orderBy(col("p_float").asc())))

File: testprojects/scala_main/pipelines/LIVY_SCALA/code/src/main/scala/io/prophecy/pipelines/livy_scala/graph/WindowFunction_1.scala
package io.prophecy.pipelines.livy_scala.graph

import io.prophecy.libs._
import io.prophecy.pipelines.livy_scala.config.ConfigStore._
import io.prophecy.pipelines.livy_scala.config.Context
import io.prophecy.pipelines.livy_scala.udfs.UDFs._
import io.prophecy.pipelines.livy_scala.udfs._
import org.apache.spark._
import org.apache.spark.sql._
import org.apache.spark.sql.functions._
import org.apache.spark.sql.types._
import org.apache.spark.sql.expressions._
import java.time._

object WindowFunction_1 {

  def apply(context: Context, in: DataFrame): DataFrame = {
    import org.apache.spark.sql.expressions.{Window, WindowSpec}
    in.withColumn(
        "variable",
        row_number().over(
          Window
            .partitionBy(col("year"),                 col("industry_code_ANZSIC"))
            .orderBy(col("industry_name_ANZSIC").asc, col("rme_size_grp").asc)
        )
      )
      .withColumn(
        "value",
        row_number().over(
          Window
            .partitionBy(col("year"),                 col("industry_code_ANZSIC"))
            .orderBy(col("industry_name_ANZSIC").asc, col("rme_size_grp").asc)
        )
      )
      .withColumn(
        "unit",
        row_number().over(
          Window
            .partitionBy(col("year"),                 col("industry_code_ANZSIC"))
            .orderBy(col("industry_name_ANZSIC").asc, col("rme_size_grp").asc)
        )
      )
  }

}
